packages/lexical-list/src/checkList.ts

```diff
@@ -0,0 +1,300 @@
+      }
+    }),
+  );
+}
+
+function handleCheckItemEvent(event: PointerEvent, callback: () => void) {
+  const target = event.target;
+
+  if (!isHTMLElement(target)) {
+    return;
+  }
+
+  // Ignore clicks on LI that have nested lists
+  const firstChild = target.firstChild;
+
+  if (
+    isHTMLElement(firstChild) &&
+    (firstChild.tagName === 'UL' || firstChild.tagName === 'OL')
+  ) {
+    return;
+  }
+
+  const parentNode = target.parentNode;
+
+  // @ts-ignore internal field
+  if (!parentNode || parentNode.__lexicalListType !== 'check') {
+    return;
+  }
+
+  const rect = target.getBoundingClientRect();
+  const pageX = event.pageX / calculateZoomLevel(target);
+  if (
+    target.dir === 'rtl'
+      ? pageX < rect.right && pageX > rect.right - 20
+      : pageX > rect.left && pageX < rect.left + 20
+  ) {
+    callback();
+  }
+}
+
+function handleClick(event: Event) {
+  handleCheckItemEvent(event as PointerEvent, () => {
+    if (isHTMLElement(event.target)) {
+      const domNode = event.target;
+      const editor = getNearestEditorFromDOMNode(domNode);
+
+      if (editor != null && editor.isEditable()) {
+        editor.update(() => {
+          const node = $getNearestNodeFromDOMNode(domNode);
+
+          if ($isListItemNode(node)) {
+            domNode.focus();
+            node.toggleChecked();
+          }
+        });
+      }
+    }
+  });
+}
+
+function handlePointerDown(event: PointerEvent) {
+  handleCheckItemEvent(event, () => {
+    // Prevents caret moving when clicking on check mark
+    event.preventDefault();
+  });
+}
+
+function getActiveCheckListItem(): HTMLElement | null {
+  const activeElement = document.activeElement;
+
+  return isHTMLElement(activeElement) &&
+    activeElement.tagName === 'LI' &&
+    activeElement.parentNode != null &&
+    // @ts-ignore internal field
+    activeElement.parentNode.__lexicalListType === 'check'
+    ? activeElement
+    : null;
+}
+
+function findCheckListItemSibling(
+  node: ListItemNode,
+  backward: boolean,
+): ListItemNode | null {
+  let sibling = backward ? node.getPreviousSibling() : node.getNextSibling();
+  let parent: ListItemNode | null = node;
+
+  // Going up in a tree to get non-null sibling
+  while (sibling == null && $isListItemNode(parent)) {
+    // Get li -> parent ul/ol -> parent li
+    parent = parent.getParentOrThrow().getParent();
+
+    if (parent != null) {
+      sibling = backward
+        ? parent.getPreviousSibling()
+        : parent.getNextSibling();
+    }
+  }
+
+  // Going down in a tree to get first non-nested list item
+  while ($isListItemNode(sibling)) {
+    const firstChild = backward
+      ? sibling.getLastChild()
+      : sibling.getFirstChild();
+
+    if (!$isListNode(firstChild)) {
+      return sibling;
+    }
+
+    sibling = backward ? firstChild.getLastChild() : firstChild.getFirstChild();
+  }
+
+  return null;
+}
+
+function handleArrownUpOrDown(
```

Comment:
```suggestion
function handleArrowUpOrDown(
```

I get that this is just copied and pasted but there's a typo in this name